async_simple/Future.h

```diff
@@ -48,14 +48,17 @@ class Promise;
 // he should call makeReadyFuture().
 template <typename T>
 class Future {
+private:
+    using inner_value_type = std::conditional_t<std::is_void_v<T>, Unit, T>;
+
 public:
-    using value_type = std::conditional_t<std::is_void_v<T>, Unit, T>;
-    Future(FutureState<value_type>* fs) : _sharedState(fs) {
+    using value_type = T;
```

Comment:
加几行注释说明下 value_type 和 inner_value_type 的意义，我觉得没有相关背景的人一看代码肯定就懵了。

Reply:
> 加几行注释说明下 value_type 和 inner_value_type 的意义，我觉得没有相关背景的人一看代码肯定就懵了。

```
    // If T is void, the inner_value_type is Unit. It will be used by `FutureState` and `LocalState`.
    // Because `Try<void>` cannot distinguish between `Nothing` state and `Value` state
    // It maybe remove Unit after next version, and then will change the `Try<void>` to 
    // distinguish between `Nothing` state and `Value` state
```
这个可以吗

Reply:
我觉得可以